src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,87 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SEPARATOR = System.lineSeparator();
+    private static final String SEPARATION_SYMBOL = ",";
+    private static final String SPACE_SYMBOL = " ";
+    private static final String OPERATION_TYPE_SUPPLY = "supply";
+    private static final String OPERATION_TYPE_BUY = "buy";
+    private static final String RESULT = "result";
+    private static final int INDEX_OPERATION = 0;
+    private static final int INDEX_NUMBER = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String textFile = readFromFile(fromFileName);
+        String report = parseDataByCategory(textFile);
+        writeToFile(report, toFileName);
+    }
+
+    private String readFromFile(String fileName) {
+        StringBuilder fileText = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String line = reader.readLine();
+            while (line != null) {
+                fileText.append(line).append(SPACE_SYMBOL);
+                line = reader.readLine();
+            }
+            System.out.println(fileText);
+            return fileText.toString();
+        } catch (IOException e) {
+            throw new RuntimeException("The file was not read: " + fileName, e);
+        }
+    }
+
+    private String parseDataByCategory(String data) {
+        int countSupply = calculateSumByCategory(data, OPERATION_TYPE_SUPPLY);
+        int countBuy = calculateSumByCategory(data, OPERATION_TYPE_BUY);
+        return getReport(countSupply, countBuy);
+    }
 
+    private int calculateSumByCategory(String data, String categoryCalculate) {
```

Comment:
```suggestion
    private int[] calculateSumByCategory(String data) {
```
can we put both result numbers in array? in order to use only 1 loop

Reply:
good catch. You can improve your solution